README.md

```diff
@@ -123,34 +123,149 @@ For protobuf-based codegen integrated with the Maven build system, you can use
 
 For protobuf-based codegen integrated with the Gradle build system, you can use
 [protobuf-gradle-plugin][]:
-```gradle
-apply plugin: 'com.google.protobuf'
+<details open>
+<summary>Groovy DSL</summary>
 
-buildscript {
-  repositories {
-    mavenCentral()
-  }
-  dependencies {
-    classpath 'com.google.protobuf:protobuf-gradle-plugin:0.8.5'
-  }
+```groovy
+plugins {
+  id "com.google.protobuf" version "0.8.8"
+  java
+}
+
+task.plugins {
+  // Add grpc output without any option.  grpc must have been defined in the
+  // protobuf.plugins block.
+  grpc { }
 }
 
 protobuf {
   protoc {
-    artifact = "com.google.protobuf:protoc:3.5.1-1"
+    // The artifact spec for the Protobuf Compiler
+    artifact = "com.google.protobuf:protoc:3.6.1"
   }
   plugins {
+    // an artifact spec for a protoc plugin, with "grpc" as
+    // the identifier, which can be referred to in the "plugins"
+    // container of the "generateProtoTasks" closure.
     grpc {
-      artifact = 'io.grpc:protoc-gen-grpc-java:1.17.1'
+      artifact = 'io.grpc:protoc-gen-grpc-java:1.18.0'
     }
   }
   generateProtoTasks {
     all()*.plugins {
-      grpc {}
+      // Apply the "grpc" plugin whose spec is defined above, without
+      // options.  Note the braces cannot be omitted, otherwise the
+      // plugin will not be added. This is because of the implicit way
+      // NamedDomainObjectContainer binds the methods.
+      grpc { }
     }
   }
 }
 ```
+</details>
+
+A community-supported Kotlin-DSL example is also available:
+<details>
+<summary>Kotlin DSL</summary>
+
+```kotlin
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+import com.google.protobuf.gradle.*
+import org.gradle.kotlin.dsl.provider.gradleKotlinDslOf
+
+buildscript {
+    extra.set("grpcVersion", "1.18.0")
+    extra.set("protocVersion", "3.6.1")
+    extra.set("kotlinVersion", "1.3.20")
+    extra.set("protoGradleVersion", "0.8.8")
+}
+
+plugins {
+    kotlin("jvm") version "1.3.20"
+    java
+    id("com.google.protobuf") version "0.8.8"
+}
+
+repositories {
+    mavenCentral()    
+    maven("https://plugins.gradle.org/m2/")
+}
+
+sourceSets {
+    main {
+        // Optional: customize where .proto are found for src/main/
+        proto {
+            // In addition to the default 'src/main/proto'
+            srcDir 'src/main/protobuf'
+            srcDir 'src/main/protocolbuffers'
+        }
+        // Optional: compile the generated Java code
+        java {
+            srcDirs(
+                "build/generated/source/proto/main/java",
+                "build/generated/source/proto/main/grpc"
+            )
+        }
```

Comment:
Why is this necessary? Generated files are automatically added as the input for Java compile tasks. Is it because it's not the case for Kotlin compile tasks?